.github/workflows/preview-cleanup.yml

```diff
@@ -1,12 +1,15 @@
 name: Preview
 
-# run when a pull request is closed
 on:
   pull_request:
     types: [closed]
 
 jobs:
   code-engine:
+    # Run the job if it's not triggered by a PR or if the PR is from the same repository
+    # This is because this job requires a secret, and secrets cannot be shared with forks
+    if: github.event.pull_request == null || github.event.pull_request.head.repo.full_name == github.repository
```

Comment:
for my own curiosity, if this is set to only run`on: pull_request` when would `github.event.pull_request == null`?

Reply:
You're right! No needed in this case!